mitodl/mitxpro
# feat: add external course id fields and enhance admin models

### What are the relevant tickets?




Initial part of https://github.com/mitodl/hq/issues/4032

### Description (What does it do?)


- #### New Fields `Course.external_course_id` and `CourseRun.external_course_run_id`

  - Adds `Course.external_course_id` and `CourseRun.external_course_run_id` to manage the external course mappings. This will help us add a consistent field with the external course data.

  - At this point, we do not have any consistent field in xPro that could be used to map the external Emeritus course data. The `Course.title` and `Course.readable_id` do not follow any specific pattern that could be used for data ingestion. Same goes for `CourseRun.courseware_id` and `CourseRun.title`.

  - Once these fields are deployed on Prod, we will manually fill the external course ids so that we use it for the data ingestion.

- #### Admin changes

  - Adds course platform to the course run list view and a platform filter to ease the manual changes.

### How can this be tested?


- Visit django admin course model, you should see a new field named `external_course_id`
- Visit django admin course run model, you should see `Platform` on the list page and a `Platform` filter as well.
- Course run detail should have a new field `extern_course_run_id`
